hydromt_wflow/wflow.py

```diff
@@ -1859,24 +1858,39 @@ def write_forcing(
                         self.write_config()  # re-write config
                         fn_out = fn_default_path
 
+            # Check if all dates between (starttime, endtime) are in all da forcing
+            start = pd.to_datetime(self.get_config("starttime"))
+            end = pd.to_datetime(self.get_config("endtime"))
+            missings = False
+            for da in self.forcing.values():
+                if "time" in da.coords:
+                    times = da.time.values
+                    if pd.to_datetime(start) < pd.to_datetime(times[0]):
+                        start = pd.to_datetime(times[0])
+                        missings = True
+                    if pd.to_datetime(end) > pd.to_datetime(times[-1]):
+                        end = pd.to_datetime(times[-1])
+                        missings = True
+                        start = pd.to_datetime(times[0]).to_pydatetime()
+                        end = pd.to_datetime(times[-1]).to_pydatetime()
```

Comment:
I suggest to remove the `to_pydatatime()` here to always keep `end` (and `start`) as datetime object and only do the conversion when you update the config on L1881